lib/project_types/script/cli.rb

```diff
@@ -47,23 +53,24 @@ module UI
   autoload :ScriptProject, Project.project_filepath('script_project')
 
   class ScriptProjectError < StandardError; end
-  class InvalidContextError < ScriptProjectError; end
-  class ScriptProjectAlreadyExistsError < ScriptProjectError; end
-  class ServiceFailureError < ScriptProjectError; end
-  class TestSuiteNotFoundError < ScriptProjectError; end
   class DependencyError < ScriptProjectError; end
   class DependencyInstallError < ScriptProjectError; end
+  class InvalidContextError < ScriptProjectError; end
   class InvalidExtensionPointError < ScriptProjectError
     attr_reader :type
     def initialize(type)
       @type = type
     end
   end
+  class ServiceFailureError < ScriptProjectError; end
+  class ScriptProjectAlreadyExistsError < ScriptProjectError; end
   class ScriptNotFoundError < ScriptProjectError
     attr_reader :script_name, :extension_point_type
     def initialize(extension_point_type, script_name)
       @script_name = script_name
       @extension_point_type = extension_point_type
     end
   end
+  class TestError < ScriptProjectError; end
+  class TestSuiteNotFoundError < ScriptProjectError; end
```

Comment:
The way we have all the Error classes here is that now we lose which layer they are for.  Is it possible to have layers/domain/errors.rb to contain all errors for that layer as a way to reduce the number of these files, but still have some categorization?

Reply:
Sure 👍 I created `error.rb` files in the layers which define a module that we can write our errors in. I made it a module so we don't have to autoload every error class in `cli.rb`.

Another option could be to just define the errors within the appropriate module in `cli.rb`. That just might make things a bit messier in that file